internal/appland/client.go

```diff
@@ -239,3 +259,29 @@ func (client *Client) DeleteAPIKey() error {
 	client.context.SetAPIKey("")
 	return nil
 }
+
+func (client *clientImpl) TestAPIKey(apiKey string) (bool, error) {
+	// Test the api key by making a request with an invalid scenario
+	// id. If the response is NotFound, the API key is valid. If the
+	// response is Unauthorized, the API key is invalid. Any other
+	// response is an error.
```

Comment:
This seems hacky - I think it'd be sufficient to test some heuristics here:
- `apiKey` is base64 URL encoded
- the decoded string can be split into two parts
- the first element of the split string is a valid `uint`
- the second element of the string looks like a UUID

Reply:
I agree that using `0` as the scenario id is hacky. I see value in making a real API call, to ensure that the API key the user enters is valid, in the same way that we validate their credentials. We can replace this with a more suitable call when one is available.

Reply:
The API key contains the numeric user id? I think it’s better to use the login name instead.

Reply:
If we make this change, then yes, we'll need to check for more dots in the decoded string. It might actually be better to pick a separator that's not a legal character in a login, though (if there is such a character).

Reply:
Seem like a fine choice.

Reply:
The choice to use the numeric ID over the login was based on performance. At 5m queries each, select on `id` was ~15% faster than the login name. It's fine if we want to switch it, I'm just providing some justification.

Reply:
We should add a unique index on login, which really needs to be there anyway for integrity purposes.

Reply:
Looks like it there is one: https://github.com/applandinc/appland/blob/master/db/schema.rb#L25

Reply:
nm, that's on the wrong side of the relationship.

I've created a Trello card to address this: https://trello.com/c/cbkgWddi. I'll put this PR on hold it's addressed.

Reply:
PR for the API key changes is here: https://github.com/applandinc/appland/pull/396